engine/orchestration/test/com/cloud/vm/VirtualMachineManagerImplTest.java

```diff
@@ -505,4 +513,24 @@ public void testSendStopWithNullAnswer() throws Exception {
 
         Assert.assertFalse(actual);
     }
+
```

Comment:
I have merged this into `4.7` cleanly and have forward merged into `4.8` without a problem.  I have a merge conflict when forward merging this code from `4.8` into `master`.

Merge conflict:
```
    @Test
<<<<<<< HEAD
    public void testExeceuteInSequence() {
        assertTrue(_vmMgr.getExecuteInSequence(HypervisorType.XenServer) == false);
        assertTrue(_vmMgr.getExecuteInSequence(HypervisorType.KVM) == false);
        assertTrue(_vmMgr.getExecuteInSequence(HypervisorType.VMware) == HypervisorGuru.VmwareFullClone.value());
        assertTrue(_vmMgr.getExecuteInSequence(HypervisorType.Ovm3) == VirtualMachineManager.ExecuteInSequence.value());
=======
    public void testCheckIfCanUpgrade() throws Exception {
        when(_vmInstance.getState()).thenReturn(State.Stopped);
        when(_serviceOfferingMock.isDynamic()).thenReturn(true);
        when(_vmInstance.getServiceOfferingId()).thenReturn(1l);
        when(_serviceOfferingMock.getId()).thenReturn(2l);

        ServiceOfferingVO mockCurrentServiceOffering = mock(ServiceOfferingVO.class);

        when(_offeringDao.findByIdIncludingRemoved(anyLong(), anyLong())).thenReturn(mockCurrentServiceOffering);
        when(mockCurrentServiceOffering.getUseLocalStorage()).thenReturn(true);
        when(_serviceOfferingMock.getUseLocalStorage()).thenReturn(true);
        when(mockCurrentServiceOffering.getSystemUse()).thenReturn(true);
        when(_serviceOfferingMock.getSystemUse()).thenReturn(true);
        when(mockCurrentServiceOffering.getTags()).thenReturn("x,y");
        when(_serviceOfferingMock.getTags()).thenReturn("z,x,y");

        _vmMgr.checkIfCanUpgrade(_vmInstance, _serviceOfferingMock);
>>>>>>> 4.8
    }
```

I suspect both functions are valid, but I want to verify with you.

Please advise...